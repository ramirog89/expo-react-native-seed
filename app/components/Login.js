import { AuthSession } from "expo";
import React from "react";
import { Image, Text, View, Button } from "react-native";
import PropTypes from "prop-types";

import jwtDecode from "jwt-decode";
import getEnvVars from "../../environment";
import { toQueryString, nonce } from "../helpers/authHelper";

import commonStyles from "../styles";
import { STATUS } from "../constants/application";

const { AUTH0 } = getEnvVars();

const Login = ({ onAuthentication }) => {
  const handleResponse = response => {
    const authResponse = {};

    if (response.error) {
      authResponse.status = STATUS.ERROR;
    } else {
      // Retrieve the JWT token and decode it
      const jwtToken = response && response.params && response.params.id_token;
      const decoded = jwtDecode(jwtToken);

      // TODO: later We could keep locally token.
      authResponse.status = STATUS.SUCCESS;
      authResponse.data = decoded;
    }

    onAuthentication(authResponse);
  };

  const login = async () => {
    // Retrieve the redirect URL, add this to the callback URL list
    // of your Auth0 application.
    const redirectUrl = AuthSession.getRedirectUrl();
    const response = await AuthSession.startAsync({
      authUrl:
        `${AUTH0.DOMAIN}/authorize` +
        toQueryString({
          client_id: AUTH0.CLIENT_ID,
          response_type: "token id_token",
          scope: "openid profile",
          nonce,
          redirect_uri: redirectUrl
        })
    });

    handleResponse(response);
  };

  return (
    <View style={commonStyles.containerMain}>
      <View style={commonStyles.welcomeContainer}>
        <Image
          source={
            __DEV__
              ? require("../assets/images/nextonlabs-logo.png")
              : require("../assets/images/nextonlabs-logo.png")
          }
          style={commonStyles.welcomeImage}
        />
      </View>
      <View style={commonStyles.getStartedContainer}>
        <Text style={commonStyles.getStartedText}>
          Welcome to React Native Starter
        </Text>
      </View>
      <View>
        <Button rounded onPress={() => login()} title="Log in" />
      </View>
    </View>
  );
};

Login.navigationOptions = {
  header: null
};

Login.propTypes = {
  onAuthentication: PropTypes.func
};

export default Login;
