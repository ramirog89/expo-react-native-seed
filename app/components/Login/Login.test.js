import React from "react";
import renderer from "react-test-renderer";

import Login from "./Login";

describe("<Login />", () => {
  it("has 1 child", () => {
    const rendered = renderer.create(<Login />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
