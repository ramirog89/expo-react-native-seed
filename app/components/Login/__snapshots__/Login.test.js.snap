// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> has 1 child 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 80,
          "marginLeft": -10,
          "marginTop": 3,
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 200,
        "marginHorizontal": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(96,100,109, 1)",
          "fontSize": 17,
          "lineHeight": 24,
          "textAlign": "center",
        }
      }
    >
      Welcome to React Native Starter
    </Text>
  </View>
  <View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log in
        </Text>
      </View>
    </View>
  </View>
</View>
`;
